astropy/io/fits/card.py

```diff
@@ -959,7 +1053,98 @@ def _format_comment(self):
         else:
             return f' / {self._comment}'
 
-    def _format_image(self):
+    @property
+    def _full_unit_regexp(self):
+        s = re.escape(self.unit_str_start)
+        l = re.escape(self.unit_str_delimiters[0])
+        r = re.escape(self.unit_str_delimiters[1])
+        e = re.escape(self.unit_str_end)
+        unit_str_position = self.unit_str_position.lower()
+        if unit_str_position in ['start', 'beginning']:
+            ms = '^'
+            me = ''
+        elif unit_str_position == 'end':
+            ms = ''
+            me = '$'
+        else:
+            raise ValueError(f'Unknown unit string position {self.unit_str_position}.  Expecting "start" or "end"')
+        return f'{ms}{s}{l}.*{r}{e}{me}'
+
+    def _get_comment_unit(self, comment):
+        """Returns `~astropy.units.Unit` found in comment, if defined, else ``None``"""
+        # Extract unit together with lead-in and delimiter
+        # https://stackoverflow.com/questions/8569201/get-the-string-within-brackets-in-python
+        m = re.search(self._full_unit_regexp, comment)
+        if m is None:
+            #log.debug(f'no unit matching "{self._full_unit_regexp}" in "{comment}"')
+            return None
+        # Strip off delemeters
```

Comment:
If you write the regex correctly, with a `(?P<unit>.*)`, you can get a group out that contains the stripped part. I'm no real expert, but see, e.g., `_zdef_re` in `io.fits.hdu.compressed` and its use.

Reply:
Thanks for the pointer!  I figured their should be a better way.  I had asked my 14 year-old son, who, is quite an accomplished programmer, "do you understand regular expressions?" and he responded "does anyone?"

Reply:
I certainly don't!!  :smiley: (and I'll check my (nearly) 14-year old roblox lua coder...)